euphonic/readers/castep.py

```diff
@@ -179,14 +190,20 @@ def read_phonon_data(
         idx = 0
         previous_qpt_id = -1
         repeated_qpt_ids = defaultdict(set)
+        loto_split_indices = set()
+
         while True:
             try:
-                qpt_id, qpt, qweight, qfreq, _ = _read_frequency_block(
-                    f, n_branches)
+                frequency_block = _read_frequency_block(f, n_branches)
+                qpt_id = frequency_block.qpt_id
             except EOFError:
```

Comment:
Presumably because end-of-file is hit. I think this is one of the more acceptable uses of exceptions as a control structure in Python; certainly less scary than relying on an IndexError.

Reply:
Revised: now returning None if the expected end-of-block was hit.